pdns/ssqlite3.cc

```diff
@@ -38,104 +39,188 @@
-  SSqlStatement* bind(const string& name, long long value) { int idx = name2idx(name); if (idx>0) { sqlite3_bind_int64(d_stmt, idx, value); }; return this; };
-  SSqlStatement* bind(const string& name, unsigned long long value) { int idx = name2idx(name); if (idx>0) { sqlite3_bind_int64(d_stmt, idx, value); }; return this; }
-  SSqlStatement* bind(const string& name, const std::string& value) { int idx = name2idx(name); if (idx>0) { sqlite3_bind_text(d_stmt, idx, value.c_str(), value.size(), SQLITE_TRANSIENT); }; return this; }
-  SSqlStatement* bindNull(const string& name) { int idx = name2idx(name); if (idx>0) { sqlite3_bind_null(d_stmt, idx); }; return this; }
-
-  SSqlStatement* execute() {
+  SSqlStatement* bind(const string& name, bool value) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_int(d_stmt, idx, value ? 1 : 0);
+    };
+    return this;
+  }
+
+  SSqlStatement* bind(const string& name, int value) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_int(d_stmt, idx, value);
+    };
+    return this;
+  }
+
+  SSqlStatement* bind(const string& name, uint32_t value) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_int64(d_stmt, idx, value);
+    };
+    return this;
+  }
+
+  SSqlStatement* bind(const string& name, long value) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_int64(d_stmt, idx, value);
+    };
+    return this;
+  }
+
+  SSqlStatement* bind(const string& name, unsigned long value) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_int64(d_stmt, idx, static_cast<sqlite3_int64>(value));
+    };
+    return this;
+  }
+
+  SSqlStatement* bind(const string& name, long long value) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_int64(d_stmt, idx, value);
+    };
+    return this;
+  };
+
+  SSqlStatement* bind(const string& name, unsigned long long value) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_int64(d_stmt, idx, static_cast<sqlite3_int64>(value));
+    };
+    return this;
+  }
+
+  SSqlStatement* bind(const string& name, const std::string& value) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_text(d_stmt, idx, value.c_str(), static_cast<int>(value.size()), SQLITE_TRANSIENT);
+    };
+    return this;
+  }
+
+  SSqlStatement* bindNull(const string& name) override
+  {
+    int idx = name2idx(name);
+    if (idx > 0) {
+      sqlite3_bind_null(d_stmt, idx);
+    };
+    return this;
+  }
+
+  SSqlStatement* execute() override
+  {
     prepareStatement();
     if (d_dolog) {
-      g_log<<Logger::Warning<< "Query "<<((long)(void*)this)<<": " << d_query << endl;
+      g_log << Logger::Warning << "Query " << this << ": " << d_query << endl;
```

Comment:
It looks like the existing code really wants to print the address of the object, probably to correlate the log line, and I _think_ this will no longer be possible with this change? I would argue this calls for a unique ID or something when logging is enabled, but this might be out of scope of this PR.